### PR TITLE
Tolerations taints code cleanup

### PR DESCRIPTION
**Pull request by ingvagabund**
Removing duplicated code. Broken down the changes into smaller steps for easier review.

@damemi @seanmalloy PTAL

### CODE REVIEW

File: pkg/descheduler/strategies/node_taint.go [hunk 1]
```diff
@@ -56,7 +51,12 @@ func deletePodsViolatingNodeTaints(client clientset.Interface, policyGroupVersio
 			if maxPodsToEvict > 0 && nodePodCount[node]+1 > maxPodsToEvict {
 				break
 			}
-			if !checkPodsSatisfyTolerations(pods[i], node) {
+			if !utils.TolerationsTolerateTaintsWithFilter(
+				pods[i].Spec.Tolerations,
+				node.Spec.Taints,
+				func(taint *v1.Taint) bool { return taint.Effect == v1.TaintEffectNoSchedule },
+			) {
+				klog.V(2).Infof("Not all taints with NoSchedule effect are tolerated after update for Pod %v on node %v", pods[i].Name, node.Name)
```
Comment (seanmalloy) [inline pkg/descheduler/strategies/node_taint.go:59]: nit: make `Pod` lower case in this log message to make it consistent with the lower case `node`

Comment (ingvagabund) [reply pkg/descheduler/strategies/node_taint.go]: addressed

**Comment by seanmalloy** [comment]
/lgtm
/assign @aveshagarwal @ravisantoshgudimetla 

@ingvagabund looks like Travis CI failed to create the kind cluster. Just need to re-run the Travis CI job, but I don't have access to trigger the job again.

https://travis-ci.org/github/kubernetes-sigs/descheduler/jobs/677873044

**Comment by ingvagabund** [comment]
Looks like it would be easier to just push --force :-|
@ravisantoshgudimetla @aveshagarwal do you have permissions to reopen the issue?

**Comment by ingvagabund** [comment]
@aveshagarwal can you try /reopen one more time?

**Pull request merged**